### PR TITLE
[INLONG-4961][DataProxy] Add dataproxy-sdk-golang

### LINKED ISSUE #4961: [Feature][DataProxy] Golang SDK
### Description

Java SDK Code：
https://github.com/apache/inlong/tree/master/inlong-sdk/dataproxy-sdk

Java SDK Example：
https://inlong.apache.org/docs/next/sdk/dataproxy-sdk/example

C++ SDK：
https://github.com/apache/inlong/issues/4960

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gunli**
### [INLONG-4961][DataProxy] add dataproxy-sdk-golang

- Fixes #4961 

### Motivation

Add a golang version SDK of InLong Data Proxy

### Modifications

- Add a discoverer to do service discovery;
- Add a conn pool to keep the connections to the server;
- Add a Client interface to handle request;
- Add a worker to do the data send/receive work;
- Add other util packages/functions to help the whole send/receive jobs;

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [x] This change added tests and can be verified as follows:

  - Prepare a groupID and some streamIDs in the server side and define the log schema of the streamIDs;
  - Write a test client to send messages to the server, example codes can be found in: dataproxy/example_test.go;
  - Verify how many message was sent and received.

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)
  - I will create a follow-up issue for adding the documentation later

### CODE REVIEW

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# dataproxy-sdk-golang
```
Comment (dockerzhang) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md:1]: This line is needless.

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+## Overview
+
+dataproxy-sdk-golang is the golang version of InLong data proxy client SDK.
+
```
Comment (dockerzhang) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md:6]: Extra blank line

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+```
+
+### Options
+
+refer: dataproxy/options.go
+
+``` go
+// Options is the data proxy go client configs
```
Comment (dockerzhang) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md:147]: data proxy 
->
DataProxy

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md (hunk 2 above)
Comment (gunli) [reply inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md]: fixed now

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md (hunk 3 above)
Comment (gunli) [reply inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/README.md]: fixed now

**Comment by dockerzhang** [comment]
Please rebase from the master to fix the failed UT after https://github.com/apache/inlong/pull/8128 is merged.

**Review by dockerzhang — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Pull request merged**